src/QMCWaveFunctions/BsplineFactory/SplineC2C.cpp

```diff
@@ -141,8 +141,8 @@ void SplineC2C<ST>::applyRotation(const ValueMatrix& rot_mat, bool use_stored_co
         // cur_elem points to the real componend of the coefficient.
         // Imag component is adjacent in memory.
         const auto cur_elem = Nsplines * i + 2 * j;
-        ST newval_r{0.};
-        ST newval_i{0.};
+        RealType newval_r{0.};
+        RealType newval_i{0.};
```

Comment:
This method doesn't look to work like the R2R one.  There the rot matrix stays double when st is float. So the multiplication is done at full precision.  Here I believe due to the assignments wr, wi the multiplication is done at lower precision and only the accumulation at full precision. So I think you want to make the type of wr and wi RealType as well.